controllers/storagecluster/obcstorageclasses.go

```diff
@@ -49,21 +49,17 @@ func (s *obcStorageClasses) ensureDeleted(r *StorageClusterReconciler, storageCl
 
 	storageClass := &storagev1.StorageClass{}
 	storageClass.Name = scName
-	err := r.Client.Get(r.ctx, client.ObjectKeyFromObject(storageClass), storageClass)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			r.Log.Info("OBC StorageClass already deleted", "Name", scName)
-			return ctrl.Result{}, nil
-		}
-		r.Log.Error(err, "failed to get OBC StorageClass", "Name", scName)
-		return ctrl.Result{}, err
-	}
 
 	if err := r.Client.Delete(r.ctx, storageClass); err != nil {
-		r.Log.Error(err, "failed to delete OBC StorageClass", "Name", scName)
-		return ctrl.Result{}, err
+		if client.IgnoreNotFound(err) != nil {
+			r.Log.Error(err, "failed to delete OBC StorageClass", "Name", scName)
+			return ctrl.Result{}, err
+		}
+		// Log if the resource was not found
+		r.Log.Info("OBC StorageClass already deleted", "Name", scName)
```

Comment:
I don't think that is extra log statement is required. Let's stick with
```
	if err := r.Client.Delete(r.ctx, storageClass); client.IgnoreNotFound(err) != nil {
```

Reply:
d@rewantsoni done + squash